### PR TITLE
fix: v9 form controls with underline should have underline-specific disabled styling

### LINKED ISSUE #24761: [Bug]: Disabled state in underline appearance variants should only have bottom border, matching its rest state
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
https://master--628d031b55e942004ac95df1.chromatic.com/?path=/docs/components-input--default
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://master--628d031b55e942004ac95df1.chromatic.com/?path=/docs/components-input--default

## Actual Behavior
Disabled state in underline appearances have a full border.

image

## Expected Behavior
Disabled state in underline appearances have only a bottom border to match its rest state. This would affect all form components with an underline appearance variant (Input, SpinButton, Select, Dropdown, Combobox).

image

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by smhigley**
Fixes #24761

Adds these changes:
- Input: switches to `borderColor` disabled styling so underline retains the single border
- Select: same as Input, with the minor variation that underline + disabled needs to specify top/left/right transparent borders to override nave `<select>` styles
- SpinButton: makes filled disabled variants be the same as outline to match Input & makes the same change for the underline style. 
- Combobox & Dropdown: fix disabled styles to match Input

### CODE REVIEW

File: change/@fluentui-react-combobox-e478d812-6a6a-4b46-88a4-6a97b5dbcfbd.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "fix: \"update disabled + underline styles, have all text-like form control disabled states match",
```
Comment (spmonahan) [inline change/@fluentui-react-combobox-e478d812-6a6a-4b46-88a4-6a97b5dbcfbd.json:3]:
```suggestion
  "comment": "fix: update disabled + underline styles, have all text-like form control disabled states match",
```

File: change/@fluentui-react-input-2a9cb459-0806-4724-975c-73942f819d43.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "fix: \"update disabled + underline styles, have all text-like form control disabled states match",
```
Comment (spmonahan) [inline change/@fluentui-react-input-2a9cb459-0806-4724-975c-73942f819d43.json:3]:
```suggestion
  "comment": "fix: update disabled + underline styles, have all text-like form control disabled states match",
```

File: change/@fluentui-react-select-ca9fbff7-be4b-431a-9be6-0f53bbc1ace7.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "fix: \"update disabled + underline styles, have all text-like form control disabled states match",
```
Comment (spmonahan) [inline change/@fluentui-react-select-ca9fbff7-be4b-431a-9be6-0f53bbc1ace7.json:3]:
```suggestion
  "comment": "fix: update disabled + underline styles, have all text-like form control disabled states match",
```

**Review by spmonahan — APPROVED** [review]

**Review by behowell — APPROVED** [review]

File: change/@fluentui-react-combobox-e478d812-6a6a-4b46-88a4-6a97b5dbcfbd.json (hunk 1 above)
Comment (smhigley) [reply change/@fluentui-react-combobox-e478d812-6a6a-4b46-88a4-6a97b5dbcfbd.json:3]: oops, this is what I get for trying to copy/paste in WSL 😅. Thanks for catching it!

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                             |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                       Change |
| :---------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr> | `76.153 kB`<br />`24.665 kB` | `78.076 kB`<br />`24.991 kB` |  `1.923 kB` <br />`326 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr> | `75.885 kB`<br />`24.635 kB` | `77.363 kB`<br />`24.893 kB` |  `1.478 kB` <br />`258 B`  |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                       |  `24.428 kB`<br />`7.899 kB` |  `24.112 kB`<br />`7.892 kB` |     `-316 B` <br />`-7 B`  |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                    |  `23.255 kB`<br />`7.954 kB` |  `23.428 kB`<br />`8.023 kB` |      `173 B` <br />`69 B`  |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                        | `45.826 kB`<br />`12.858 kB` |  `44.42 kB`<br />`12.815 kB` | `-1.406 kB` <br />`-43 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `62.941 kB`<br />`17.678 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.907 kB`<br />`52.983 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.446 kB`<br />`11.033 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d8dec99eae6f1138d8dade6ec57663916a348762'>d8dec99eae6f1138d8dade6ec57663916a348762</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/Avatar-mount_base.html">1777</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/Avatar-mount.html">1736</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/Button-mount_base.html">1196</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/Button-mount.html">1215</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/FluentProvider-mount_base.html">2055</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/FluentProvider-mount.html">2114</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/FluentProviderWithTheme-mount_base.html">747</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/FluentProviderWithTheme-mount.html">748</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">680</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">704</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">750</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">760</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/MakeStyles-mount_base.html">2439</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/MakeStyles-mount.html">2483</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/SpinButton-mount_base.html">3547</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25543/perf-test-react-components/SpinButton-mount.html">3486</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Pull request merged**